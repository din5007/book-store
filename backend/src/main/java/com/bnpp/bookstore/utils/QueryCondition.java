package com.bnpp.bookstore.utils;

import io.micrometer.common.util.StringUtils;
import org.springframework.stereotype.Component;

@Component("queryUtil")
public class QueryCondition {

  public boolean checkNull(String field) {
    return StringUtils.isBlank(field);
  }
}
