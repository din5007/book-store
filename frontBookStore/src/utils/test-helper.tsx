import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { BrowserRouter } from "react-router-dom";


function customRender(component : any , store?: any) {

    return render(<StoreProvider store={store}>
        <BrowserRouter>
            {component}
        </BrowserRouter>
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;