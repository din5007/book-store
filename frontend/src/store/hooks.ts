import { createTypedHooks } from "easy-peasy";

import { AppStoreModel } from ".";

export const { useStoreActions, useStoreState } = createTypedHooks<AppStoreModel>();