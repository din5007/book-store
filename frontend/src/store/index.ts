import { HttpService } from "../service/http/http.service";
import { createStore } from 'easy-peasy';
import { BookModel, booksModel } from "./books/books.model";
import { cartModel, CartModel } from "./cart/cart.model";
import { userModel, UserModel } from "./users/user.model";
import { OrderModel, orderModel } from "./order/order.model";

export interface AppStoreModel {
    books: BookModel;
    cart: CartModel;
    user: UserModel;
    order: OrderModel;
}

export const appStoreModel : AppStoreModel = {
    books: booksModel,
    cart: cartModel,
    user: userModel,
    order: orderModel,
}

export function storeModel() {
    return createStore(appStoreModel, {
        injections: {
            httpService: new HttpService()
        },
        initialState: {
            
        },
    })
}
export const store =  storeModel();