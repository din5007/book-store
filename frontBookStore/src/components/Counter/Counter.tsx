import { FC, useState } from "react";
import { Button, Card } from "react-bootstrap";
import { useStoreActions } from "../../store/hooks";

const Counter : FC<{id: number, quantity: number}> = ({ id, quantity }) => {
    const { addToCart, decrementQuantity } = useStoreActions((actions) => actions.cart);
    const [count, setCount] = useState(quantity);

    return (
        <>
        <Button type="button" size="sm" disabled={count == 1} onClick={() => { 
            decrementQuantity({  id });
            setCount(prev => prev - 1);
          }}>-</Button>
          <Card.Text>&nbsp;{count}&nbsp;</Card.Text>
          <Button type="button" size="sm" disabled={count == 10} onClick={() => {
            addToCart({  id });
            setCount(prev => prev + 1);
          }}>+</Button>
        </>
    );
}
export default Counter;