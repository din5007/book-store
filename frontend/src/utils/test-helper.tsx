import { StoreProvider } from "easy-peasy";
import { render } from '@testing-library/react';
import { AppStoreModel, store } from "../store";
import { Router } from 'react-router-dom';
import { ReactNode } from "react";



// const provider : React.FC<{children : ReactNode, store: any}> = ({ children, store }) => {
//     return (
        
//     )
// }

function customRender(component : any , store: any) {

    return render(<StoreProvider store={store}>
        {component}
    </StoreProvider>);
}

export * from '@testing-library/react';
export { customRender } ;