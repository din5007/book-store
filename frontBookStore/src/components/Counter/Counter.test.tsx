import { createStore } from 'easy-peasy';
import { customRender as render } from '../../utils/test-helper';
import Counter from "./Counter"
import { appStoreModel } from '../../store';
import { describe, expect, it, vi } from 'vitest';

const store = createStore(appStoreModel, {
    injections: {
        httpservice : {
            get: vi.fn()
        }
    },
    initialState: {

    }
})

describe('Counter', () => {
    it('Counter should load properly with given params - snapshot', () => {
        const {container}  = render(<Counter id={10} quantity={100}/>, store);
        expect(container).toMatchSnapshot()
    });
})