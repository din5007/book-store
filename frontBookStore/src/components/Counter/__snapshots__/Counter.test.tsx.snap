// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Counter > Counter should load properly with given params - snapshot 1`] = `
<div>
  <button
    class="btn btn-primary btn-sm"
    type="button"
  >
    -
  </button>
  <p
    class="card-text"
  >
     
    100
     
  </p>
  <button
    class="btn btn-primary btn-sm"
    type="button"
  >
    +
  </button>
</div>
`;
